packages/process/src/node/terminal-process.ts

```diff
@@ -79,72 +79,100 @@ export class TerminalProcess extends Process {
         }
         this.logger.debug('Starting terminal process', JSON.stringify(options, undefined, 2));
 
-        try {
-            this.terminal = spawn(
-                options.command,
-                (isWindows && options.commandLine) || options.args || [],
-                options.options || {}
-            );
-
-            process.nextTick(() => this.emitOnStarted());
-
-            // node-pty actually wait for the underlying streams to be closed before emitting exit.
-            // We should emulate the `exit` and `close` sequence.
-            this.terminal.on('exit', (code, signal) => {
-                // Make sure to only pass either code or signal as !undefined, not
-                // both.
-                //
-                // node-pty quirk: On Linux/macOS, if the process exited through the
-                // exit syscall (with an exit code), signal will be 0 (an invalid
-                // signal value).  If it was terminated because of a signal, the
-                // signal parameter will hold the signal number and code should
-                // be ignored.
-                if (signal === undefined || signal === 0) {
-                    this.onTerminalExit(code, undefined);
-                } else {
-                    this.onTerminalExit(undefined, signame(signal));
-                }
-                process.nextTick(() => {
-                    if (signal === undefined || signal === 0) {
-                        this.emitOnClose(code, undefined);
-                    } else {
-                        this.emitOnClose(undefined, signame(signal));
+        const startTerminal = (command: string): { terminal: IPty | undefined, inputStream: Writable } => {
+            try {
+                return this.createPseudoTerminal(command, options, ringBuffer);
+            } catch (error) {
+                // Normalize the error to make it as close as possible as what
+                // node's child_process.spawn would generate in the same
+                // situation.
+                const message: string = error.message;
+
+                if (message.startsWith('File not found: ') || message.endsWith(NodePtyErrors.ENOENT)) {
+                    if (isWindows && command && !command.toLowerCase().endsWith('.exe')) {
+                        const commandExe = command + '.exe';
+                        this.logger.warn(`Trying terminal command '${commandExe}' because '${command}' was not found.`);
```

Comment:
I don't have a strong opinion here, but imo this could be an info message as well. 
I guess it depends on whether we see this more as a feature or more as a fallback for a misconfiguration

Reply:
Actually, that's a good point. I might rather lean to `'debug'` as this is supposed to be unsurprising and totally supported behaviour, but should still be recorded in case some flow doesn't happen as expect. For example, in the — decidedly inadvisable — case where a `.exe` file exists but the developer doesn't want it to be used in the case that the unsuffixed file doesn't exist?

Reply:
Changed in commit b5d02c6fba7400b3ee358601891d713af8225569 to debug level.